typing/mode.ml

```diff
@@ -2460,8 +2460,26 @@ module Modality = struct
     let zap_to_floor = function
       | Const c -> c
       | Undefined -> Misc.fatal_error "modality Undefined should not be zapped."
-      | Exactly (_, m) ->
-        let c = Mode.zap_to_floor m in
+      | Exactly (mm, m) ->
+        let m = Mode.zap_to_floor m in
```

Comment:
I’m afraid that after staring at this comment for several minutes, I still can’t quite get to the bottom of it.

Here is a start at rewriting it that would resonate with me more:

```
The goal here is to find some [c] such that [meet_with c] can replace [Exactly(mm, m)]. For soundness, we require [m <= meet_with c mm]: that the result of [meet_with c mm] must be weaker than [m]. For completeness, we require [meet_with c mm <= m]: that the result of [meet_with c mm] is at least as useful as [m].

We will find the highest [c] that can satisfy completeness, and then prove that such a [c] is sound.

We thus start with [meet_with c mm <= m], which we rewrite to [c <= imply_with mm m]. [m] has already been zapped to its floor (this is in [zap_to_floor], after all), so that’s a constant. But [mm] is not. We must zap it. But in which direction? …
```

and this is where I lose the trail of the argument. There is some property about `imply_with` that you’re relying on… what is it? And then the comment says find a ceil with `mm’ >= mm`, but below is a call `mm’ = zap_to_floor mm`, which is in opposition.

My goal here isn’t a beautifully crafted comment, exactly. That’s not worth it. But I do think we should be able to create a shared understanding of what this function is meant to do and why, and this comment seems the best way of doing so.

Reply:
I improved the comment as beautiful as I can. 😄 
I also think again and decided to take `mm`'s ceil instead of zapping it to floor (reasons explained). The changes don't change behavior currently since all modules's mode are fixed already.

Reply:
Ah, yes, excellent. I now fully understand this and like the explanation. Thanks!